src/main/java/org/hiero/sketch/table/FullMembership.java

```diff
@@ -64,18 +62,58 @@ public IMembershipSet sample(final int k, final long seed) {
         return this.sampleUtil(randomGenerator, k);
     }
 
+    @Override
+    public IMembershipSet union (final IMembershipSet otherSet) throws NullArgumentException {
+        if (otherSet == null)
+            throw new NullArgumentException("Can not perform union with a null");
+        if (otherSet instanceof FullMembership)
+            return new FullMembership(Integer.max(this.rowCount, otherSet.getSize()));
+        return otherSet.union(this);
+    }
+
+    @Override
+    public IMembershipSet intersection (final IMembershipSet otherSet) throws NullArgumentException {
+        if (otherSet == null)
+            throw new NullArgumentException("Can not perform intersection with a null");
+        if (otherSet instanceof FullMembership)
+            return new FullMembership(Integer.min(this.rowCount, otherSet.getSize()));
+        return otherSet.intersection(this);
+    }
+
+    @Override
+    public IMembershipSet setMinus (final IMembershipSet otherSet) throws NullArgumentException {
+        if (otherSet == null)
+            throw new NullArgumentException("Can not perform setMinus with a null");
+        if (otherSet instanceof FullMembership) {
```

Comment:
No, this case does not call the isMember method. The isMember method may potentially use a predicate (in the Lazy case) and that might be more costly.